### PR TITLE
feat: navigate to next lesson

### PR DESCRIPTION
**Pull request by Chifez**
## What does this PR do?
This PR helps to navigate to the next lesson 

Fixes # (place issue number here without bracket)

https://github.com/rotimi-best/classroomio/assets/69228399/37f8a57d-fc70-4571-b50e-2d40107d17c3

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [X] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change adds a new database migration
- [ ] This change requires a documentation update

## How should this be tested?

- run on localhost

### Required

- [ ] Filled out the "How to test" section in this PR
- [ ] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand bits
- [ ] Ran `pnpm build`
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the ClassroomIO Docs if changes were necessary

### CODE REVIEW

File: apps/dashboard/src/routes/courses/[id]/lessons/[...lessonParams]/+page.svelte [hunk 1]
```diff
@@ -373,6 +423,18 @@
...
+        disabled={disabled.prev}
+        class={`px-2 my-2 pr-4 border-t-0 border-b-0 border-l-0 border border-gray-300 flex items-center ${
+          disabled.prev && 'cursor-not-allowed'
+        }`}
+        on:click={goToPrevLesson}
+      >
+        <ChevronLeft size={24} />
+        {#if !$isMobile}
+          <span> Prev </span>
+        {/if}
```
Comment (rotimi-best) [inline apps/dashboard/src/routes/courses/[id]/lessons/[...lessonParams]/+page.svelte]: You only use js for things that can't be done in css.

just use css to hide this element on mobile

File: apps/dashboard/src/routes/courses/[id]/lessons/[...lessonParams]/+page.svelte [hunk 2]
```diff
@@ -408,6 +470,16 @@
...
         {/if}
       </button>
+      <button
+        disabled={disabled.next}
+        class={`px-2 my-2 flex items-center ${disabled.next && 'cursor-not-allowed'}`}
+        on:click={goToNextLesson}
+      >
+        {#if !$isMobile}
+          <span> Next </span>
+        {/if}
```
Comment (rotimi-best) [inline apps/dashboard/src/routes/courses/[id]/lessons/[...lessonParams]/+page.svelte]: same here, use css

File: apps/dashboard/src/routes/courses/[id]/lessons/[...lessonParams]/+page.svelte [hunk 3]
```diff
@@ -240,6 +246,50 @@
...
+
+  $: {
+    const currentLessonIdx = $lessons.findIndex((lesson) => lesson.id === data.lessonId);
+    disabled.next =
+      currentLessonIdx === -1 ||
+      $lessons.slice(currentLessonIdx + 1).every((lesson) => !lesson.is_unlocked);
+    disabled.prev =
+      currentLessonIdx === -1 ||
+      $lessons.slice(0, currentLessonIdx).every((lesson) => !lesson.is_unlocked);
+  }
```
Comment (rotimi-best) [inline apps/dashboard/src/routes/courses/[id]/lessons/[...lessonParams]/+page.svelte]: This is a complex logic.

Just delete this reactivity and write one function `isNextOrPrevDisabled`

```svelte
const isNextOrPrevDisabled = (lessonId: string, isPrev: boolean) => {
  const index = $lessons.findIndex((lesson) => lesson.id === lessonId);

  return isPrev ? !$lessons[index - 1] : !$lessons[index + 1]
}

<button
  disabled={isNextOrPrevDisabled(data.lessonId, true)}
>
Prev
</button>

<button
  disabled={isNextOrPrevDisabled(data.lessonId)}
>
Next
</button>
```

File: apps/dashboard/src/routes/courses/[id]/lessons/[...lessonParams]/+page.svelte [hunk 4]
```diff
@@ -240,6 +246,50 @@
...
+
+  const goToPrevLesson = () => {
+    const currentLessonIdx = $lessons.findIndex((lesson) => lesson.id === data.lessonId);
+
+    if (currentLessonIdx !== -1 && currentLessonIdx - 1 >= 0) {
+      const prevLesson = $lessons[currentLessonIdx - 1];
+      console.log('next lesson', prevLesson);
+
+      fetchReqData(prevLesson.id, true);
+      const prevLessonUrl =
```
Comment (rotimi-best) [inline apps/dashboard/src/routes/courses/[id]/lessons/[...lessonParams]/+page.svelte]: All these logic are so complex. and you have a ton of repeated code.

The logic below is built upon the previous comment I made about the reactivity.

```ts
const goToNextOrPrevLesson = (lessonId: string, isPrev: boolean) => {
  const isDisabled = isNextOrPrevDisabled(lessonId, isPrev);

  // Always use early return
  if (isDisabled) return;
  
  const index = $lessons.findIndex((lesson) => lesson.id === lessonId);
  const nextOrPrevLesson = isPrev ? $lessons[index - 1] : $lessons[index + 1];

  const isLocked = $globalStore.isStudent && !nextOrPrevLesson.is_unlocked;
  // Early return for locked lesson
  if (isLocked) return;

  const path = `/courses/${$course.id}/lessons/${nextOrPrevLesson.id}`;
  // fetchReqData(nextLesson.id, true); - I don't know why you need this though
  goto(path);
}
```

File: apps/dashboard/src/routes/courses/[id]/lessons/[...lessonParams]/+page.svelte [hunk 5]
```diff
@@ -408,6 +470,16 @@
           <CheckmarkOutlineIcon size={24} class="carbon-icon" />
         {/if}
       </button>
+      <button
+        disabled={disabled.next}
+        class={`px-2 my-2 flex items-center ${disabled.next && 'cursor-not-allowed'}`}
+        on:click={goToNextLesson}
+      >
+        {#if !$isMobile}
+          <span> Next </span>
```
Comment (rotimi-best) [inline apps/dashboard/src/routes/courses/[id]/lessons/[...lessonParams]/+page.svelte]: Why is there no translation @Chifez ???😟

File: apps/dashboard/src/lib/components/CourseContainer/index.svelte [hunk 6]
```diff
@@ -61,6 +61,7 @@
       isPermitted = false;
     }
   }
+  $: console.log('this is the lesson', $lessons);
```
Comment (rotimi-best) [inline apps/dashboard/src/lib/components/CourseContainer/index.svelte]: Please remove.

**Review by rotimi-best — CHANGES_REQUESTED** [review]
@Chifez Do you review your PR before making it open?

There are at least 2 comments that I shouldn't have made if you paid attention and reviewed your PR.

File: apps/dashboard/src/routes/courses/[id]/lessons/[...lessonParams]/+page.svelte (hunk 3 above)
Comment (Chifez) [reply apps/dashboard/src/routes/courses/[id]/lessons/[...lessonParams]/+page.svelte:291]: i needed to handle an extra case when the next lesson being navigated to is currently locked, the function only navigates to the nex function that is not locked

**Review by rotimi-best — APPROVED** [review]
LGTM

**Pull request merged**